pkg/controller/metrics/metrics.go

```diff
@@ -192,9 +206,18 @@ func (mc *Calculator) Start(stopCh <-chan struct{}) error {
 					MetricClusterDeploymentDeprovisioningUnderwaySeconds.WithLabelValues(
 						cd.Name,
 						cd.Namespace,
-						GetClusterDeploymentType(&cd)).Set(
+						clusterType).Set(
 						time.Since(cd.CreationTimestamp.Time).Seconds())
 				}
+
+				var value float64
+				if paused, err := strconv.ParseBool(cd.Annotations[constants.SyncsetPauseAnnotation]); err == nil && paused {
+					value = 1.0
+				}
+				metricClusterDeploymentSyncsetPaused.WithLabelValues(
+					cd.Name,
+					cd.Namespace,
+					clusterType).Set(value)
```

Comment:
This is a great idea and thanks for sending us the PR directly!

We want to limit the data we report as much as possible, as implemented it's reporting 0's for every cluster in the system, I think you can just skip the 0's.

Then you would want to make sure to delete any metric that shouldn't be set (for clusters that were paused but no longer are) so it stops getting reported (and eventually cycles out of the prometheus data as it expires a week or two later), which you can do idempotently with this approach:
https://github.com/openshift/hive/blob/master/pkg/controller/clusterdeployment/clusterdeployment_controller.go#L1927

Reply:
That makes sense, and thanks for the quick review.  I'll rework as you suggest.